v2/metadata/client.go

```diff
@@ -131,11 +131,32 @@ func (c *Client) GetPipeline(ctx context.Context, pipelineName string, pipelineR
 		return nil, err
 	}
 
+	err = c.putParentContexts(ctx, &pb.PutParentContextsRequest{
+		ParentContexts: []*pb.ParentContext{{
+			ChildId:  pipelineRunContext.Id,
+			ParentId: pipelineContext.Id,
+		}},
+	})
+	if err != nil {
+		return nil, err
+	}
+
 	return &Pipeline{
 		pipelineCtx:    pipelineContext,
 		pipelineRunCtx: pipelineRunContext,
 	}, nil
+}
 
+func (c *Client) putParentContexts(ctx context.Context, req *pb.PutParentContextsRequest) error {
+	_, err := c.svc.PutParentContexts(ctx, req)
+	if err != nil {
+		if status.Convert(err).Code() == codes.AlreadyExists {
+			// already exists code is expected when multiple requests are sent in parallel
```

Comment:
Can we add a logging here which indicates that the parental edge already exists

Reply:
This will happen too many times, it's probably not worth logging.
(Because every task runs in parallel, only the first task will successfully put the parent edge, all other tasks will get this already exists status code)